src/core/integrators/velocity_verlet_inline.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef INTEGRATORS_VELOCITY_VERLET_HPP
+#define INTEGRATORS_VELOCITY_VERLET_HPP
+
+#include "ParticleRange.hpp"
+#include "config.hpp"
+#include "particle_data.hpp"
+#include "rotation.hpp"
+
+inline void velocity_verlet_propagate_vel_pos(const ParticleRange &particles) {
+
+  auto const skin2 = Utils::sqr(0.5 * skin);
+  for (auto &p : particles) {
+#ifdef ROTATION
+    propagate_omega_quat_particle(p);
+#endif
+
+// Don't propagate translational degrees of freedom of vs
+#ifdef VIRTUAL_SITES
+    if (p.p.is_virtual)
+      continue;
+#endif
+    for (int j = 0; j < 3; j++) {
+#ifdef EXTERNAL_FORCES
+      if (!(p.p.ext_flag & COORD_FIXED(j)))
+#endif
+      {
+        /* Propagate velocities: v(t+0.5*dt) = v(t) + 0.5 * dt * a(t) */
+        p.m.v[j] += 0.5 * time_step * p.f.f[j] / p.p.mass;
+
+        /* Propagate positions (only NVT): p(t + dt)   = p(t) + dt *
+         * v(t+0.5*dt) */
+        p.r.p[j] += time_step * p.m.v[j];
+      }
+    }
+
+    /* Verlet criterion check*/
+    if (Utils::sqr(p.r.p[0] - p.l.p_old[0]) +
+            Utils::sqr(p.r.p[1] - p.l.p_old[1]) +
+            Utils::sqr(p.r.p[2] - p.l.p_old[2]) >
+        skin2)
+      set_resort_particles(Cells::RESORT_LOCAL);
+  }
+}
+
+inline void
+velocity_verlet_propagate_vel_final(const ParticleRange &particles) {
```

Comment:
it would also be nice to have a docstring mentioning which part of VV is performed here (maybe a reference to literature)